matrix-sdk/src/main/java/org/matrix/androidsdk/db/MXMediasCache.java

```diff
@@ -310,7 +310,8 @@ private String downloadableUrl(String url, int width, int height) {
                 return mContentManager.getDownloadableUrl(url);
             }
         } else {
-            return url;
+            // do not allow non-mxc content URLs: we should not be making requests out to whatever http urls people send us
+            return null;
```

Comment:
This is quite dangerous to return null, as before it was never returning null.
I think in some cases this should provocate NPE.
Please check all the calls to this method.
Please also use `@Nullable` annotation to the return type and update the Javadoc.

-> Maybe this is not the best way to fix this issue

Reply:
You're right, I missed some calls of this method which involve NPE